Mage/src/main/java/mage/game/combat/Combat.java

```diff
@@ -1659,6 +1664,25 @@ public UUID getDefenderId(UUID attackerId) {
      * @return
      */
     public UUID getDefendingPlayerId(UUID attackingCreatureId, Game game) {
+        return getDefendingPlayerId(attackingCreatureId, game, false);
+    }
+
+    /**
+     * Returns the playerId of the player that is attacked by given attacking
+     * creature or formerly-attacking creature.
+     *
+     * @param attackingCreatureId
+     * @param game
+     * @return
+     */
+    public UUID getDefendingPlayerId(UUID attackingCreatureId, Game game, boolean allowFormer) {
+        if (allowFormer) {
+            return Stream.concat(groups.stream(), formerGroups.stream())
+                    .filter(group -> (group.getAttackers().contains(attackingCreatureId) || group.getFormerAttackers().contains(attackingCreatureId)))
+                    .map(CombatGroup::getDefendingPlayerId)
+                    .findFirst()
+                    .orElse(null);
+        }
```

Comment:
What about helper methods like `attacksAlone`, `noAttackers` and `isPlaneswalkerAttacked` -- is it can be affected same rules? `isPlaneswalkerAttacked` can be affected cause it's included in "defending player" definition

![shot_241222_215839](https://github.com/user-attachments/assets/6b97acaf-bd1e-47c2-9b8c-52d4b7fda9f1)

Reply:
P.S. Planeswalker code used in one card -- Oath of Kaya. It's not affected by 802.2a (cause trigger check actual data only), but card bugged by itself.

Reply:
attacksAlone():
- SourceAttackingAloneCondition and similar conditions, used (so far) exclusively for "{this} can't be blocked as long as it's attacking alone" effects; if it's been removed from combat, it being unblockable is irrelevant.
- AttacksAlone...TriggeredAbility, are using it in `checkTrigger` on event type `GameEvent.EventType.ATTACKER_DECLARED` which, I believe, happens during SBAs on declaring them as attackers so it's (nearly?) impossible for the attacker to have been removed from combat before the function is used - you'd have to have a creature dying as a state-based action consequence of being declared as an attacker.

noAttackers():
- Used to skip Declare Blockers and Combat Damage steps if there are no attacking creatures.